openpgp/keys.go

```diff
@@ -477,16 +498,31 @@ func addSubkey(e *Entity, packets *packet.Reader, pub *packet.PublicKey, priv *p
 			// packets that are in the wrong place (like misplaced 0x13 signatures)
 			// until we get to one that works.  For a test case,
 			// see TestWithBadSubkeySignaturePackets.
+
 			continue
 		}
-		subKey.Sig = sig
-		err = e.PrimaryKey.VerifyKeySignature(subKey.PublicKey, subKey.Sig)
+		err = e.PrimaryKey.VerifyKeySignature(subKey.PublicKey, sig)
 		if err != nil {
-			return errors.StructuralError("subkey signature invalid: " + err.Error())
+			// Non valid signature, so again, no need to abandon all hope, just continue;
+			// make a note of the error we hit.
+			lastErr = errors.StructuralError("subkey signature invalid: " + err.Error())
+			continue
+		}
+		switch sig.SigType {
+		case packet.SigTypeSubkeyBinding:
+			subKey.Sig = sig
```

Comment:
Maybe a little safer to stick with the first-writer-wins semantics that the library had before?